File: zerojudge/a005-eva-的回家作業/solution.py
"""
* Author: 1chooo<hugo970217@gmail.com>
* Problem link: https://zerojudge.tw/ShowProblem?problemid=a005
* Status: AC
"""

if __name__ == "__main__" :

    run_time = int(input())

    for i in range(0, run_time) :
        input_string = str(input()).split(" ")

        a = int(input_string[0])
        b = int(input_string[1])
        c = int(input_string[2])
        d = int(input_string[3])

        if ((b - a) == (c - b)) :
            print(a, b, c, d, d + (b - a))
        else :
            print(a, b, c, d, int(d * (b / a)))

File: zerojudge/a005-eva-的回家作業/solution.cpp
/*
 * Author: 1chooo<hugo970217@gmail.com>
 * Problem link: https://zerojudge.tw/ShowProblem?problemid=a005
 * Status: AC
 */

#include <iostream>

using namespace std;

int main() {
    int runTimes;
    cin >> runTimes;

    for (int i = 0; i < runTimes; i++) {
        int num[4];

        for (int j = 0; j < 4; j++) {
            cin >> num[j];
            cout << num[j] << " ";
        }

        if ((num[1] - num[0]) == (num[2] - num[1])) {
            cout << num[3] + (num[1] - num[0]) << endl;
        } else {
            cout << num[3] * (num[1] / num[0]) << endl;
        }
    }
    return 0;
}
